strapi/strapi
# [v5] Run dialect initializations on every new connection in the pool

### What does it do?

- Use knex pool.afterCreate to run the dialect initializations instead of calling it in db constructor
- add search path to postgres initialization

### Why is it needed?

1) db constructor is sync and is returning a connection before it has been initialized
2) only the first connection in a knex pool has the initialization run; any new connections do not have it, meaning that things like sqlite foreign keys and postgres schema path are not set on those connections

### How to test it?

- all databases should still work fine as before
- now, postgres schema path should work everywhere even on raw queries that do not manually provide the schema name
- sqlite and mysql should also now have their dialect initializations run on each new connection in the pool (can test it by running a config query, then running a raw query to check if that value has been set; prior to this PR the answer would be no but now it should work)

### Related issue(s)/PR(s)

too many to find, but anything relating to postgres schemas or database initialization
